modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -138,7 +139,12 @@ export class Transaction extends BaseTransaction {
     if (this.atomTransaction.signature) {
       this.addSignature(Buffer.from(this.atomTransaction.signature).toString('hex'));
     }
-    this._type = utils.getTransactionTypeFromTypeUrl(this._atomTransaction.sendMessages[0].typeUrl);
+    const typeUrl = this.atomTransaction.sendMessages[0].typeUrl;
+    const transactionType = utils.getTransactionTypeFromTypeUrl(typeUrl);
+    if (transactionType === undefined) {
```

Comment:
would `if (!transactionType)` be cleaner or do we want to only check for undefined ?

Reply:
This is tricky with enums(learned it the hard way 😅 ), first entry in the enum has value 0 which is falsy. So condition `if(!transactionType)` would resolve to true for TransactionType.Send

Reply:
Ah I see